### DISCUSSION TITLE
Using Latex template with quarto books

### DISCUSSION THREAD

**Discussion by abduazizR** [quarto-dev/quarto-cli#9309] [Q&A]
### Description

Hi 

I am planning to use quarto books to format my dissertation. However, I cannot find a straightforward way to integrate the latex formatting with quarto

This is the Latex formatting developed by one of the students [https://github.com/arindam-bose/uic-ece-thesis-template/tree/master/uicthesis](https://github.com/arindam-bose/uic-ece-thesis-template/tree/master/uicthesis)

My question is, how can I adapt this to a quarto book so it will format everything as required by my school?

I tried the following

I downloaded [this folder](https://github.com/arindam-bose/uic-ece-thesis-template/tree/master/uicthesis) and added it to the main directory of the project and edited my  `_quarto.yml` to be like this

```
project:
  type: book
  execute-dir: project

execute: 
  echo: false
  warning: false
  message: false

book:
  title: "my dissertation title" 
  author: "Author"
  output-file: "author_diss"
  chapters:
    - index.qmd
    - dedication.qmd
    - 01-intro.qmd
    - 02-study1.qmd
    - 03-study2.qmd
    - references.qmd

bibliography: references.bib
csl: apa.csl

format:
  pdf:
    documentclass: book
    pdf-engine: pdflatex
    classoption: ["oneside", "open=any"]
    margin-left: "1in"
    margin-right: "1in"
    margin-top: "1in"
    margin-bottom: "1in" 
    mainfont: Arial
    linestretch: 2
    template: "uicthesis/thesisphd.tex"
    include-in-header: 
      - text: | 
          \usepackage{tocbibind}
          \usepackage{booktabs, caption, longtable, colortbl, array}
          \usepackage{indentfirst}
          \setlength{\parindent}{1cm}
          \usepackage{fvextra}
          \DefineVerbatimEnvironment{Highlighting}{Verbatim}{breaklines,commandchars=\\\{\}}
          \usepackage{setspace}
          \doublespacing
          \pagenumbering{gobble} 
          \usepackage{sectsty}
          \usepackage{titlesec}
          \titleformat{\chapter}[display]
          {\normalfont\bfseries}{}{0pt}{\fontsize{12}{15}\selectfont}
          \titlespacing*{\chapter}{0pt}{0ex plus 1ex minus .2ex}{4.3ex plus .2ex}
          \titlespacing*{\section}{0pt}{5.5ex plus 1ex minus .2ex}{4.3ex plus .2ex}
          \titlespacing*{\subsection}{0pt}{5.5ex plus 1ex minus .2ex}{4.3ex plus .2ex}
          \sectionfont{\fontsize{12}{15}\selectfont}
          \subsectionfont{\fontsize{12}{15}\selectfont}
          
    toc: true
    toc-depth: 3
    lot: true
    lof: true
    number-sections: TRUE
   
```

I got this error message

```
Rendering PDF
running pdflatex - 1
  This is pdfTeX, Version 3.141592653-2.6-1.40.26 (TeX Live 2024) (preloaded format=pdflatex)
   restricted \write18 enabled.
  entering extended mode
  
updating tlmgr

updating existing packages
finding package for bin/uicthesi.cls
ERROR: 
compilation failed- no matching packages
LaTeX Error: File `bin/uicthesi.cls' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: cls)

Enter file name: 
! Emergency stop.
<read *> 
         
l.6 ^^M
```

Hope you can help me with this

Thanks

**Comment by cderv** [comment]
You need to learn a bit more on how LaTeX works, and also what Quarto templates are. 

First: `uicthesis/thesisphd.tex` is not a template to be used with Quarto. This is an example from the repo your linked on how to use the template. Quarto templates are Pandoc's template with addition of partials feature to help modify part of it. Read more at https://quarto.org/docs/journals/templates.html

Second: Usually in LaTeX, templating work will happen through customized document class, and in your example you have `\documentclass{bin/uicthesi}`. So I believe you need to set the custom document class for quarto use. This is LaTeX knowledge required so that you know what to look for as configuration in quarto: https://quarto.org/docs/output-formats/pdf-basics.html#document-class

If you don't know what a document class is, you should look into it. 

Then your template defines some helpers 
````
% helpers
\input{bin/packages.tex}
\input{bin/settings.tex}
\input{bin/symbols.tex} % version 2.0.11 compatible
````
I don't know what they are but you probably need to inline them, or try including them by using modified partials or template, or providing them as includes: https://quarto.org/docs/output-formats/pdf-basics.html#latex-includes

Your template also have a customized frontmatter: https://github.com/arindam-bose/uic-ece-thesis-template/blob/480f0fb6093fd7bd6fb8e1da3bdf0123ad5c604e/uicthesis/thesisphd.tex#L22-L32
````
% Declarations for front matter
\title{Title of Your Thesis}
\author{Author's Name}
\pdegrees{Author's degree, year}
\degree{Doctor of Philosophy in Electrical and Computer Engineering}
\committee{Member 1, Chair and Advisor \\ 
	Member 2 \\ 
	Member 3 \\ 
	Member 4 \\ 
	Member 5, Affiliation}
\makeatletter
````
So there you will need to modified the right partials for Quarto: https://quarto.org/docs/journals/templates.html#latex-partials

And probably you should adapt for after the body content to. 

Best thing IMO would be to try create a custom format for Quarto based on this template you have: https://quarto.org/docs/extensions/formats.html

See example for formats in https://quarto.org/docs/extensions/listing-formats.html

Hope this helps understand - happy doc reading !
